src/python-frontend/function_call_expr.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <python-frontend/python_converter.h>
+#include <python-frontend/type_handler.h>
+#include <python-frontend/symbol_id.h>
+#include <util/expr.h>
+#include <nlohmann/json.hpp>
+
+enum class FunctionType
+{
+  Constructor,
+  ClassMethod,
+  InstanceMethod,
+  FreeFunction,
+};
+
+class function_call_expr
+{
+public:
+  function_call_expr(const nlohmann::json &call, python_converter &converter);
+
+  exprt build();
+
+private:
+  void build_function_id();
```

Comment:
Can you add some documentation for these methods? You can follow the same patterns as we have on https://github.com/esbmc/esbmc/blob/master/src/goto-symex/goto_symex_state.h.